### PR TITLE
Encapsulate the agent new version safe check, full unit test coverage

### PR DESCRIPTION
**Pull request by aleksmaus**
## What is the problem this PR solves?

Addresses comment on previous PR:
https://github.com/elastic/fleet-server/pull/683#issuecomment-909984620

## How does this PR solve the problem?

* Extract the get agent new version into a function
* Add full unit test coverage for that part

## Checklist

- [x] I have commented my code, particularly in hard-to-understand areas

## Related issues

- Relates https://github.com/elastic/fleet-server/pull/683#issuecomment-909984620

### CODE REVIEW

File: internal/pkg/model/ext.go [hunk 1]
```diff
@@ -25,3 +25,16 @@ func (m *Server) Time() (time.Time, error) {
 func (m *Server) SetTime(t time.Time) {
 	m.Timestamp = t.Format(time.RFC3339Nano)
 }
+
+// GetNewVersion returns Agent version if it is different from ver, otherwise return empty string
+func (m *Agent) GetNewVersion(ver string) string {
+	if m == nil {
```
Comment (ruflin) [inline internal/pkg/model/ext.go:31]: Is this an expected scenario or should we log an error in this case?

File: internal/pkg/model/ext.go [hunk 2]
```diff
@@ -25,3 +25,16 @@ func (m *Server) Time() (time.Time, error) {
 func (m *Server) SetTime(t time.Time) {
 	m.Timestamp = t.Format(time.RFC3339Nano)
 }
+
+// GetNewVersion returns Agent version if it is different from ver, otherwise return empty string
+func (m *Agent) GetNewVersion(ver string) string {
```
Comment (ruflin) [inline internal/pkg/model/ext.go:30]: Nit: It works also for older versions and only new :-)

**Review by ruflin — APPROVED** [review]
LGTM

**Review by scunningham — APPROVED** [review]

File: internal/pkg/model/ext.go (hunk 2 above)
Comment (aleksmaus) [reply internal/pkg/model/ext.go:30]: > Nit: It works also for older versions and only new :-)
yeah might be the best name for the function, open to suggestions.

File: internal/pkg/model/ext.go (hunk 1 above)
Comment (aleksmaus) [reply internal/pkg/model/ext.go:31]: it's not expected, the function is currently called only after successfully authenticated agent (agent is normally not nil). just additional nil safety check.

Comment (ruflin) [reply internal/pkg/model/ext.go:31]: `CheckDifferentVersion` or `GetDifferentVersion` -> took what you have in the description :-)

**Review by blakerouse — APPROVED** [review]
Looks good. Thanks for cleaning it up more and adding tests.

File: internal/pkg/model/ext.go (hunk 1 above)
Comment (aleksmaus) [reply internal/pkg/model/ext.go:31]: ```CheckDifferentVersion``` sounds good, will update

**Pull request merged**